lib/bosh/BoshDirectorClient.js

```diff
@@ -169,6 +183,16 @@ class BoshDirectorClient extends HttpClient {
     return this.request(requestDetails, expectedStatusCode);
   }
 
+  checkAndUpdateCache(deploymentName, config) {
+    return this
```

Comment:
The looping should not be from caller, but from within here. Loop through each of the director configs and get deployment. Also use `Promise.any` so that you fire get deployment calls in parallel to all directors and whoever returns first will update the cache.